platform/macos/src/MGLMapViewDelegate.h

```diff
@@ -255,6 +255,22 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (BOOL)mapView:(MGLMapView *)mapView shapeAnnotationIsEnabled:(MGLShape *)annotation;
 
+/**
+ Returns a Boolean value indicating whether the annotation should be brought into view
+ in response to selection by clicking.
+ 
+ If the delegate does not implement this method, the default value is `YES`.
+ 
+ Note that user-initiated selections are always animated.
+ 
+ @param mapView The map view that has selected the annotation.
+ @param annotation The object representing the shape annotation.
+ @return A Boolean value indicating whether the annotation should be moved into view. This value
+ can be considered to be the same as the `expose` parameter passed to `-[MGLMapView selectAnnotation:animated:expose:]`
+ */
+- (BOOL)mapView:(MGLMapView *)mapView shouldExposeAnnotationInResponseToUserSelection:(id <MGLAnnotation>)annotation;
```

Comment:
It isn’t clear to the developer how this method differs from disabling an annotation via `MGLAnnotationImage.enabled`, `MGLAnnotationView.enabled`, or `-[MGLMapViewDelegate mapView:shapeAnnotationIsEnabled]`. If the problem with these three options is that the annotation still swallows the tap gesture, then that would be a simple bug to fix rather than an API to introduce. If the problem with these three options is that the developer still wants the annotation to be “selected”, then they should investigate managing their own annotation selection state via a property in their map view delegate.

If the developer expects to use this method to implement a callout view other than the built-in one, they should instead use the custom callout view API.